addons/io_scene_gltf2/blender/exp/gltf2_blender_extract.py

```diff
@@ -105,9 +105,6 @@ def convert_swizzle_scale(scale, export_settings):
 def decompose_transition(matrix, export_settings):
     translation, rotation, scale = matrix.decompose()
 
-    # Put w at the end.
-    rotation = Quaternion((rotation[1], rotation[2], rotation[3], rotation[0]))
```

Comment:
What are the implications of removing this?  Are we sure it's the right thing to do?

Reply:
It was needed for consistency. The `convert_swizzle_rotation` function requires the quaternion to not have the components mixed. Also, it makes more sense in general to not change the order of the components until when converting from a quaternion to a JSON array, since changing the order of the components breaks quaternion multiplication (multiplication by 1 = `Quaternion((1, 0, 0, 0))` is the identity, and multiplication by k = `Quaternion((0, 0, 0, 1))` isn't).